File: Data Structures & Algorithms/eating-bananas/submission-1.java
class Solution {
    public int minEatingSpeed(int[] piles, int h) {
        int l = 1;
        int r = Arrays.stream(piles).max().getAsInt();
        int res = 1;

        while(l <= r) {
            int k = (l + r) / 2;
            int totalHours = 0;
            for(int pile : piles) {
                totalHours += (int) Math.ceil((double) pile / k);
            }        

            if(totalHours <= h) {
                res = k;
                r = k - 1;
            } else {
                l = k + 1;
            }
        }
        return res;
    }
}

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-1.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        l = 0
        res = 0
        charMap = defaultdict(int)

        for r in range(len(s)):
            charMap[s[r]] += 1
            while (r - l + 1) - max(charMap.values()) > k:
                charMap[s[l]] -= 1
                l += 1
            res = max(res, r - l + 1)
        
        return res
